be/src/services/vocabulary.service.ts

```diff
@@ -18,17 +19,44 @@ export class VocabularyService {
             _id: null,
             created: new Date(),
             lastModified: new Date()
-        }).then((result) => result.insertedId)
-          .then(id => this.getVocabular(id));
+        })
+            .then((result) => result.insertedId)
+            .then(id => this.getVocabular(id));
     }
 
     public getVocabular(id: string | ObjectId): Promise<Vocabulary> {
         return this.collection.findOne({_id: new ObjectId(id)}).then(x => <Vocabulary>x);
     }
 
+    public async deleteVocab(id: string | ObjectId, date: Date): Promise<boolean> {
+
+        if (isNaN(date.getDate())) {
```

Comment:
The [getDate()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate) returns the day of the month according to local time. Not the best number of a date for verification, please use:
```suggestion
        if (isNaN(date.getTime())) {
```

Reply:
Can a `Date` with an illegal value even exists? I'd expect it to be `null` in such case...

Reply:
The given string will always be converted into a Date type object. if it is converted successful, we will receive a normal Date, otherwise we receive `invalid date` object. Therefore, to check if it's valid, it's enough to check if the conversion to time is a number or not.